src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -405,13 +428,26 @@ private async Task ExecutePrimaryAnalysisTaskAsync(AnalysisScope analysisScope,
 
             if (WhenInitializedTask.IsFaulted)
             {
-                OnDriverException(WhenInitializedTask, this.analyzerExecutor, analysisScope.Analyzers);
+                OnDriverException(WhenInitializedTask, this.AnalyzerExecutor, analysisScope.Analyzers);
             }
             else if (!WhenInitializedTask.IsCanceled)
             {
-                this.analyzerExecutor = this.analyzerExecutor.WithCancellationToken(cancellationToken);
+                this.AnalyzerExecutor = this.AnalyzerExecutor.WithCancellationToken(cancellationToken);
+
+                var task = ProcessCompilationEventsAsync(analysisScope, analysisStateOpt, usingPrePopulatedEventQueue, cancellationToken);
+
+#if DEBUG
+                // If not using pre-populated event queue (batch mode), then verify all symbol end actions were processed.
+                if (!usingPrePopulatedEventQueue)
+                {
+                    task = task.ContinueWith(t =>
+                        {
+                            AnalyzerManager.VerifyAllSymbolEndActionsExecuted();
+                        }, cancellationToken, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+                }
```

Comment:
Why isn't this just a single statement `AnalyzerManager.VerifyAllSymbolEndActionsExecuted();` (and leave the await on the call to `ProcessCompilationEventsAsync`) ?

Reply:
Yes.